tests/integration/repos_test.go

```diff
@@ -114,30 +114,51 @@ func TestRepositories_EditBranches(t *testing.T) {
 		t.Fatalf("Repositories.GetBranch() returned error: %v", err)
 	}
 
-	if *branch.Protection.Enabled {
+	if *branch.Protected {
 		t.Fatalf("Branch %v of repo %v is already protected", "master", *repo.Name)
 	}
 
-	branch.Protection.Enabled = github.Bool(true)
-	branch.Protection.RequiredStatusChecks = &github.RequiredStatusChecks{
-		EnforcementLevel: github.String("everyone"),
-		Contexts:         &[]string{"continous-integration"},
-	}
-	branch, _, err = client.Repositories.EditBranch(*repo.Owner.Login, *repo.Name, "master", branch)
-	if err != nil {
-		t.Fatalf("Repositories.EditBranch() returned error: %v", err)
-	}
-
-	if !*branch.Protection.Enabled {
-		t.Fatalf("Branch %v of repo %v should be protected, but is not!", "master", *repo.Name)
-	}
-	if *branch.Protection.RequiredStatusChecks.EnforcementLevel != "everyone" {
-		t.Fatalf("RequiredStatusChecks should be enabled for everyone, set for: %v", *branch.Protection.RequiredStatusChecks.EnforcementLevel)
-	}
-
-	wantedContexts := []string{"continous-integration"}
-	if !reflect.DeepEqual(*branch.Protection.RequiredStatusChecks.Contexts, wantedContexts) {
-		t.Fatalf("RequiredStatusChecks.Contexts should be: %v but is: %v", wantedContexts, *branch.Protection.RequiredStatusChecks.Contexts)
+	protectionRequest := &github.ProtectionRequest{
+		RequiredStatusChecks: &github.RequiredStatusChecks{
+			IncludeAdmins: github.Bool(true),
+			Strict:        github.Bool(true),
+			Contexts:      &[]string{"continuous-integration"},
+		},
+		RequiredPullRequestReviews: &github.RequiredPullRequestReviews{
+			IncludeAdmins: github.Bool(true),
+		},
+		Restrictions: &github.BranchRestrictionsRequest{
+			Users: &[]string{*repo.Owner.Login},
+			Teams: &[]string{"team"},
+		},
+	}
+
+	protection, _, err := client.Repositories.UpdateBranchProtection(*repo.Owner.Login, *repo.Name, "master", protectionRequest)
+
+	if err != nil {
```

Comment:
Remove the blank line between the `protection, _, err := ...` line and `if err != nil {`. That way it's more visible that the error value is being checked. The blank line between those 2 lines is not helpful.